#740 rails 7 0

danlaffan opened the pull request.

jaredbeck commented on .rubocop.yml:
```diff
@@ -12,7 +12,7 @@ AllCops:
   #
   # Should be same as `ruby-version` in `.github/workflows/test.yml`
-  TargetRubyVersion: 2.4
+  TargetRubyVersion: 2.7
```
Please target 2.6, which does not reach EoL until 2022-03-31

jaredbeck commented on authlogic.gemspec:
```diff
@@ -18,16 +18,16 @@ require "authlogic/version"
   s.license = "MIT"
 
-  s.required_ruby_version = ">= 2.4.0"
+  s.required_ruby_version = ">= 2.7.0"
```
Please target 2.6, which does not reach EoL until 2022-03-31

jaredbeck commented on authlogic.gemspec:
```diff
@@ -18,16 +18,16 @@ require "authlogic/version"
-  s.add_dependency "activesupport", [">= 5.2", "< 6.2"]
+  s.add_dependency "activemodel", [">= 5.2", "<= 7.0"]
+  s.add_dependency "activerecord", [">= 5.2", "<= 7.0"]
+  s.add_dependency "activesupport", [">= 5.2", "<= 7.0"]
```
Should we specify `"< 7.1"`, so that e.g. `7.0.1` is allowed?

danlaffan replied:
I agree to all the changes.

jaredbeck reviewed (changes requested):
This is a great start! Nice work

danlaffan commented:
@jaredbeck I have updated the strategy matrix to handle Rails 5.2, 6.0, 6.1 and 7.0 using Ruby 2.6 and 2.7.  I've updated the exceptions to exclude 2.6/7.0.

I've pushed the changes but the diffs shown above don't appear updated to me.  Is there some other step I need to perform to update this PR or do I need to create a new PR?  Cheers!

jaredbeck commented:
Thanks. Nice job for your first OSS PR. Best of luck with your new app.

danlaffan commented:
@jaredbeck Yes - my first OSS contribution.  I feel like it's a big day.  Thank you very much for your help.  I really want to use Authlogic in my new app so this unblocks the road for me.

morgoth commented:
@jaredbeck Since the Rails 7 RC1 is out, can we please have it released in the stable version of gem? https://weblog.rubyonrails.org/2021/12/6/Rails-7-0-rc-1-released/

danlaffan commented:
@morgoth @jaredbeck I am happy to try this out.

danlaffan commented:
See this pull request: https://github.com/binarylogic/authlogic/pull/745
